lib/core/src/Cardano/Pool/Metrics.hs

```diff
@@ -211,12 +231,20 @@ data ErrListStakePools
      | ErrListStakePoolsErrNetworkTip ErrNetworkTip
 
 newStakePoolLayer
-     :: DBLayer IO
-     -> NetworkLayer IO t block
-     -> Trace IO Text
-     -> StakePoolLayer IO
+    :: DBLayer IO
+    -> NetworkLayer IO t block
+    -> Trace IO StakePoolLayerMsg
+    -> StakePoolLayer IO
 newStakePoolLayer db@DBLayer{..} nl tr = StakePoolLayer
     { listStakePools = do
+        lift $ logTrace tr MsgListStakePoolsBegin
+        stakePools <- rankKnownPools
```

Comment:
**[remark]** 

You have used to forbidden word (i.e. `rank`) :scream: 
It makes things clearer though :+1: